### DISCUSSION TITLE
[OSPP Week 5] Finish the rest of the work of last week and make RLDatasets.jl ready for registration

### DISCUSSION THREAD

**Discussion by Mobius1D** [JuliaReinforcementLearning/ReinforcementLearning.jl#432] [Archive]
Last week my work was:

To extend the package to d4rl datasets and google research dqn atari replay datasets.

My goal this week would be to finish the rest of the work in #413 and make the package ready for registration.

**Comment by pilgrimygy** [comment]
Well. But now we need the function proposed by Jun:
```julia
gen_dataset("JuliaRL-CartPole-BasicDQN", policy, env)
```
Because of lacking MuJoCo license, I only train the model by d4rl dataset but there is no way to test it. So we need to train algorithms in some simple env supported by RLEnv.jl, such as CartPole, Pendulum. What do you think?

**Reply by Mobius1D** [reply]
Sorry for the delayed response.

Hmm, currently I am working on lazy loading of RL unplugged datasets which is taking some time. 

But if Mujoco license is an issue, you can use d4rl-pybullet datasets. That comes for free. And for generating toy datasets, you can collect the experiences via a hook right? But that feature will soon be added soon using  RL unplugged bsuite datasets, that covers simple toy experiments and is pretty standard for testing.

In fact beyond the scope of this project, we can provide wrapped envs that work out of the box so that the users need not wrap the envs to work with the datasets.

**Reply by pilgrimygy** [reply]
Great.
